Added haptic priority system

EricBAndrews opened the pull request:
<!-- 
Thank you for making a pull request! 
Since we are very busy with getting Mlem into a releaseable state, we had to introduce this short questionnaire to help us review PRs.
Before you submit your PR, please take a few minutes to fill out all the needed information.

Please note that if you do not fill out the checklist, your PR will be automatically rejected unless you are an approved contributor. 
We apologize, as we would love to dedicate the time it deserves to every PR, but at present, we are under considerable time pressure.
-->

# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #319 
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
This PR introduces a haptic priority system, allowing users to choose whether they want to feel *all* the haptics, only the core haptics, or no haptics at all.

JakeShirley reviewed (approved):
The concept makes sense to me but I'm not sure I understand what "core" vs "all" means in this context.

EricBAndrews commented:
> The concept makes sense to me but I'm not sure I understand what "core" vs "all" means in this context.

Fair point--I named them based on how they should appear in settings, but that had the unfortunate effect of making a `some` haptic higher-priority than an `all` haptic. I've updated the *enum* names to `.sentinel, .high, .low`, which should make much more sense to use, and added some logic in the `label` field of the enum to convert them to user-friendly names on the front end.
